lib/elixir/lib/range.ex

```diff
@@ -47,6 +48,12 @@ defmodule Range do
       false
 
   """
+  # @spec range?(term) :: boolean
+  @spec range?(arg) :: false when arg: %Range{first: nil, last: nil}
+  @spec range?(arg) :: true when arg: %Range{}
+  @spec range?(term) :: false 
```

Comment:
If that works, then you have a false negative because if the function being
called accidentally returns an atom, it won't fail.

-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*